org.eclipse.m2e.core/src/org/eclipse/m2e/core/internal/embedder/EclipseLogger.java

```diff
@@ -143,4 +143,14 @@ public Logger getChildLogger(String name) {
   public String getName() {
     return Messages.EclipseLogger_name;
   }
+
+  private String toMessageString(String msg, Throwable t) {
+    if(t == null || t.getMessage() == null) {
+      return msg;
+    }
+    if(msg == null) {
+      return t.getMessage();
+    }
+    return msg + " " + t.getMessage();
```

Comment:
For log messages it is usually good practice to only pass a format string and its arguments separately so the string object creation can be avoided if the message is eventually not logged.

Reply:
I read your words but I don't understand what I can do about it.

Reply:
As this is a 1:1 as before I think its actually fine here, all calls are already guarded by a check if the log level is enabled so it will be logged with no doubt.

Reply:
You're probably not referring to avoiding `t.getMessage()` in case the message is produced lazily, are you?

Reply:
Also I don't see an overload in slf4j with a format string *and* a throwable.

Reply:
> I read your words but I don't understand what I can do about it.

I was thinking about something like https://www.slf4j.org/manual.html#typical_usage

But Christoph is right, the calls to this method are already guarded (Sorry I oversaw that at my phone).